library/core/src/marker.rs

```diff
@@ -923,3 +923,14 @@ mod copy_impls {
     #[stable(feature = "rust1", since = "1.0.0")]
     impl<T: ?Sized> Copy for &T {}
 }
+
+/// A common trait implemented by all function pointers.
+#[unstable(feature = "fn_ptr_trait", issue = "none", reason = "recently added")]
+#[lang = "fn_ptr_trait"]
+#[cfg(not(bootstrap))]
+#[rustc_deny_explicit_impl]
+pub trait FnPtr: Copy + Clone {
```

Comment:
Could this be used usable as specialization trait?

Reply:
for `min_specialization`? yes, and unlike specializing on `Copy` and `Clone` it's actually completely sound :3 this trait is implemented for all function pointer types, completely ignoring regions.